core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchModelImpl.java

```diff
@@ -245,6 +241,36 @@ private String getSearchConfigJson() {
 
     private Locale getLocale() {
         return Optional.ofNullable(resource.adaptTo(ResourceWrapper.class)).map(ResourceWrapper::getLocale)
-                .orElse(Locale.getDefault());
+                       .orElse(Locale.getDefault());
+    }
+
+    private ConnectionFailedAlert resolveConnectionFailedAlert() {
+        if (searchCAConfigurationModel == null) {
+            log.warn("Can not resolve context aware search configuration model.");
```

Comment:
Mmh, this should be already an error, not only a warning. What do you think? And I am thinking if we really need to do this check as it should never be null --> null pointer exception

Reply:
I agree, it should never be null, but the method contract of **adaptTo** is suggesting a nullable return. If I remove it, sonarlint would most probably report it.